fix: LEAP-999: Fix E2E tests on LSO

juliosgarbi opened the pull request:
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)


### Describe the reason for change
The E2E tests have been consistently failing on the LSO environment. Initial investigations revealed two primary issues affecting the stability and success rate of these tests:

**Timeout Duration**: Previously set at 20 minutes, this duration proved insufficient, leading to premature test terminations especially notable in tests that parallel those in the LSF submodules where the timeout is set at 45 minutes.

#### What does this fix?
This PR addresses the above issues by implementing the following changes:

**Increased Timeout**: The timeout for tests has been adjusted from 20 minutes to 45 minutes to align with the LSF submodule settings and accommodate tests that require longer execution times.
**Set build to production**: Set NODE_ENV to production and BUILD_NO_MINIMIZATION to false so it prevent the tests to fail; the reason: development mode loaded some task at the beginning, messing with the LSF instance in tests.

### What level of testing was included in the change?
_(check all that apply)_
- [x] e2e
- [ ] integration
- [ ] unit

juliosgarbi committed:
fix: LEAP-999: Fix E2E tests on LSO

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8648018514

juliosgarbi committed:
check if store already exists and return to not change the store

juliosgarbi commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8648660563)
> Successfully merged:  4 files changed, 52 insertions(+), 7 deletions(-)

juliosgarbi committed:
Merge branch 'develop' into 'fb-leap-999'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8648660563

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8648686234

juliosgarbi committed:
check if store is already defined

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8648983400

juliosgarbi committed:
remove the store removal

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8649286764

juliosgarbi committed:
revert changes

juliosgarbi committed:
remove headless from CI

juliosgarbi commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8650039411)
> Successfully merged:  7 files changed, 37 insertions(+), 22 deletions(-)

juliosgarbi committed:
Merge branch 'develop' into 'fb-leap-999'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8650039411

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8650063937

juliosgarbi committed:
change headless config

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8650379107

juliosgarbi committed:
add back headless'

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8651042751

juliosgarbi committed:
add more time on waitForAction

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8651278054

juliosgarbi committed:
changing the timeout for e2e tests to 30 minutes

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8651665734

juliosgarbi committed:
change the timeout to e2e test for 40 minutes

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8652075069

juliosgarbi committed:
change the timeout to 45 minutes

juliosgarbi committed:
remove left over console.log

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8652143057

juliosgarbi committed:
change the timeout for 55 minutes

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8654424701

juliosgarbi committed:
set yarn test e2e timeout to 45

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8655549409

juliosgarbi committed:
change the name to make tests easier

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8663298339

juliosgarbi committed:
add some wait to make the test pass

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8663519255

juliosgarbi committed:
add 1 more wait

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8663766305

juliosgarbi committed:
hide error layer

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8664041202

juliosgarbi committed:
revert multiple autio test

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8664175802

juliosgarbi committed:
move taxonomy tests to make tests easier

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8665244545

juliosgarbi committed:
add wait and change the text title

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8665350957

juliosgarbi committed:
remove validation on test

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8665616245

juliosgarbi committed:
change from open taxonomy by click to open by search

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8665724793

juliosgarbi committed:
add wait object to be ready

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8665885662

juliosgarbi committed:
remove all validations

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8666011206

juliosgarbi committed:
add back validations and add wait

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8666096653

juliosgarbi committed:
add wait to fix tests on FB

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8666262451

juliosgarbi committed:
add 1 one wait

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8666375469

juliosgarbi committed:
add a wait to fix tests on FB

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8666537152

juliosgarbi committed:
add wait

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8666650971

juliosgarbi committed:
fix the way that wait is being added

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8666781390

juliosgarbi committed:
move back taxonomy tests

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8666918071

bmartel commented:
>Added Waits: Additional wait times have been introduced in the taxonomy tests to ensure **adequate time for location validation**. This change has been tested on Feature branches and has shown to prevent the previously encountered failures by allowing proper loading and processing times.

@juliosgarbi Do we have a more tactical means to identify if the taxonomy is loaded and ready? This could possibly introduce a flaky test if there is not a way to guarantee the test can proceed safely. Also what is meant by the bolded part? Does that mean there is some validation that happens on load? Is this asynchronously loaded from a remote endpoint, or is it something we have loaded and controlled from the local test instance?

hlomzik commented:
Yea, I would also wait against random `wait`s with magic numbers, especially given that they added before LS init! Meaning that there is nothing on the page at this moment, so it waits for nothing. Orrrr... If there is some default task loaded every time we open LSF standalone, then we can replace this task to something lightweight.

Gondragos commented on web/libs/editor/tests/e2e/tests/taxonomy.test.js:
```diff
@@ -34,36 +34,35 @@ Scenario('Lines overlap', async ({ I, LabelStudio, AtTaxonomy }) => {
   <Text name="text" value="$text"/>
   <Taxonomy name="taxonomy" toName="text">
-    <Choice value="target group">
+    <Choice value="target group 2">
```
The changes like that could potentially break the test logic. Target group title most probably should be the same in each config.

juliosgarbi committed:
add env production and roll back the other changes

juliosgarbi commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8707030530)
> Successfully merged:  328 files changed, 99 insertions(+), 16 deletions(-)

juliosgarbi committed:
Merge branch 'develop' into 'fb-leap-999'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8707030530

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8707073603

juliosgarbi committed:
change waitforAction to 0

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8707219972

juliosgarbi committed:
add build no monimization

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8707366297

juliosgarbi committed:
fix indentation on taxonomy test

juliosgarbi committed:
Update feature-flags.ts

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8707632035

juliosgarbi committed:
change waitForAction to 300 when it headless

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8708128076

juliosgarbi committed:
Update index.html

juliosgarbi committed:
remove #react-refresh-overlay from being hidden

juliosgarbi commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8724452622)
> Successfully merged:  7 files changed, 65 insertions(+), 10 deletions(-)

juliosgarbi committed:
Merge branch 'develop' into 'fb-leap-999'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8724452622

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8724485632

juliosgarbi commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8755025167)
> Successfully merged:  43 files changed, 774 insertions(+), 789 deletions(-)

juliosgarbi committed:
Merge branch 'develop' into 'fb-leap-999'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8755025167

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8755059268
